project-level config for analyses (#8882)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/behavior-changes.md

When `require_corrected_analysis_fqns` is set to `True`, dbt:
- Routes analysis configurations from the `analyses` block in `dbt_project.yml`, enabling project-level and folder-level configurations to take effect.
- Removes the extra FQN segment so that analysis FQNs are consistent with other resource types (for example, `your_project.subdirectory.analysis_name` instead of `your_project.analyses.subdirectory.analysis_name`).

To enable project-level analysis configuration, set the flag to `True` and add an `analyses` block in your `dbt_project.yml`. Note that `enabled` is the only config supported for project-level configuration of analyses. For example:

<File name='dbt_project.yml'>

```yaml
flags:
  require_corrected_analysis_fqns: true

analyses:
  +enabled: true  # enable all analyses by default
```
</File>

You can also configure analyses at the folder level by nesting subfolders under your project name in the `analyses` block.

<File name='dbt_project.yml'>

> ```yaml

Reviewer: ```suggestion
```yaml
flags:
  require_corrected_analysis_fqns: true

```

For more information, refer to [Data test configurations](/reference/data-test-configs).

### Project-level configuration for analyses <Lifecycle status="beta" />

:::info Beta feature
The project-level configuration for analyses is a beta feature in <Constant name="core" /> v1.12.
:::

Previously, project-level configuration for [analyses](/docs/build/analyses) in `dbt_project.yml` was silently ignored. Fully qualified names (FQNs) for analyses also contained an extra `analyses` path segment that was inconsistent with other resource types.

> When `require_corrected_analysis_fqns` is set to `True`, dbt:

Reviewer 2: Changing capitalization to match YAML standard
```suggestion
When `require_corrected_analysis_fqns` is set to `true`, dbt:
```

- Routes analysis configurations from the `analyses` block in `dbt_project.yml`, enabling project-level configurations to take effect.

> - Removes the extra FQN segment so that analysis FQNs are consistent with other resource types (for example, `your_project.subdirectory.analysis_name` instead of `your_project.analyses.subdirectory.analysis_name`).

Reviewer 2: Not a huge deal but using same example as in upgrade guide.
```suggestion
- Removes the extra FQN segment so that analysis FQNs are consistent with other resource types (for example, `your_project.my_analysis` instead of `your_project.analyses.my_analysis`).
```

For more information, refer to [Data test configurations](/reference/data-test-configs).

### Project-level configuration for analyses <Lifecycle status="beta" />

:::info Beta feature
The project-level configuration for analyses is a beta feature in <Constant name="core" /> v1.12.
:::

Previously, project-level configuration for [analyses](/docs/build/analyses) in `dbt_project.yml` was silently ignored. Fully qualified names (FQNs) for analyses also contained an extra `analyses` path segment that was inconsistent with other resource types.

When `require_corrected_analysis_fqns` is set to `True`, dbt:

> - Routes analysis configurations from the `analyses` block in `dbt_project.yml`, enabling project-level and folder-level configurations to take effect.

Reviewer 3: ```suggestion
- Routes analysis configurations from the `analyses` block in `dbt_project.yml`, enabling project-level configurations to take effect.
```

> - Removes the extra FQN segment so that analysis FQNs are consistent with other resource types (for example, `your_project.subdirectory.analysis_name` instead of `your_project.analyses.subdirectory.analysis_name`).

Reviewer 3: does this make it so model names and analysis names have to be unique? or has that always been the case @sriramr98

<AnalysesProjectLevelConfig />

<File name='dbt_project.yml'>

```yaml
flags:
  require_corrected_analysis_fqns: true

analyses:
  +enabled: false  # disable all analyses by default
```
</File>

> You can also configure analyses at the folder level by nesting subfolders under your project name in the `analyses` block.

Reviewer 3: again, this is how all project-level configs work so i don't think we need a separate callout for it here
